script.js

```diff
@@ -0,0 +1,146 @@
+// Possible states of the game
+const GameStates = {
+    INPROGRESS: "In Progress",
+    XVICTORY: "X wins",
+    OVICTORY: "O wins",
+    TIE: "Tie"
+}
+let gameState = GameStates.INPROGRESS
+
+const board = document.querySelectorAll('.grid-square')
+
+// Possible states of a square
+const SquareStates = {
+    X: "X",
+    O: "O",
+    EMPTY: "Empty"
+}
+
+const isTaken = (squareId) => {
+    return isX(squareId) || isO(squareId)
+}
+
+const isX = (squareId) => {
+    return !document.getElementById('ximg' + squareId).classList.contains('hidden')
+}
+
+const isO = (squareId) => {
+    return !document.getElementById('oimg' + squareId).classList.contains('hidden')
+}
+
+const getSquareStates = (squareId) => {
+    if (isX(squareId)){
+        return SquareStates.X
+    }
+    else if (isO(squareId)){
+        return SquareStates.O
+    }
+    else {
+        return SquareStates.EMPTY
+    }
+}
+
+const getTurn = () => {
```

Comment:
I think it might make more sense to store a global turn variable of type integer that gets incremented whenever someone moves